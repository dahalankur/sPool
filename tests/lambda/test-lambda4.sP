# a lambda with all statement types

(quack -> quack) it =  lambda quack ():
    1 + 3 / 2 * 5 + -2
    a = 7
    int b = 4
    def store int myFunc (int y): return y;
    while (a > 2):
        a = a - 1;
    if (a < 2):
        print("hello")
    else
        if (a > 0): 
            print("good");
        print("too small");
    return ;
