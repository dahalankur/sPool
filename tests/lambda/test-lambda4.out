PROGRAM {[ 
    DEFINE(ARROWTY(QUACKTY -> QUACKTY), it, LAMBDA(QUACKTY, FORMALS(), [ 
        BINOP(BINOP(INT(1), PLUS, BINOP(BINOP(INT(3), DIV, INT(2)), TIMES, INT(5))), PLUS, UNOP(NEG, INT(2))) 
        ASSIGN(a, INT(7)) 
        DEFINE(INTTY, b, INT(4)) 
        FUN(STORE, INTTY, myFunc, FORMALS((INTTY, y)), [ 
            RETURN(VAR(y))]) 
        WHILE(BINOP(VAR(a), GREATER, INT(2)), [ 
            ASSIGN(a, BINOP(VAR(a), MINUS, INT(1)))]) 
        IF(BINOP(VAR(a), LESS, INT(2)), [ 
            CALL(print, ARGS(STRING("hello")))], [ 
            IF(BINOP(VAR(a), GREATER, INT(0)), [ 
                CALL(print, ARGS(STRING("good")))], []) 
            CALL(print, ARGS(STRING("too small")))]) 
        RETURN(NOEXPR)]))]}
